accounts/abi/bind/backends/simulated.go

```diff
@@ -72,37 +73,48 @@ type SimulatedBackend struct {
 	filterSystem *filters.FilterSystem // for filtering database logs
 
 	config *params.ChainConfig
+
+	coinbase common.Address
 }
 
 // NewSimulatedBackendWithDatabase creates a new binding backend based on the given database
 // and uses a simulated blockchain for testing purposes.
 // A simulated backend always uses chainID 1337.
-func NewSimulatedBackendWithDatabase(database ethdb.Database, alloc core.GenesisAlloc, gasLimit uint64) *SimulatedBackend {
+func NewSimulatedBackendWithDatabase(database ethdb.Database, alloc core.GenesisAlloc, gasLimit uint64) (*SimulatedBackend, error) {
 	genesis := core.Genesis{
 		Config:   params.AllEthashProtocolChanges,
 		GasLimit: gasLimit,
 		Alloc:    alloc,
 	}
-	blockchain, _ := core.NewBlockChain(database, nil, &genesis, nil, ethash.NewFaker(), vm.Config{}, nil, nil)
 
+	return NewSimulatedBackendWithDatabaseAndGenesis(database, genesis)
+}
+
+func NewSimulatedBackendWithDatabaseAndGenesis(database ethdb.Database, genesis core.Genesis) (*SimulatedBackend, error) {
```

Comment:
I feel like we should add a config for simulation backend. For example:

```go
type Config struct {
    Genesis    core.Genesis,
    VmConfig vm.Config
    // etc
}

NewSimulatedBackendWithConfig()
```

Reply:
would you mind if this is done in a seprate PR?